{
	"document": {
		"aggregate_severity": {
			"namespace": "https://www.suse.com/support/security/rating/",
			"text": "moderate"
		},
		"category": "csaf_security_advisory",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright 2024 SUSE LLC. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "Security update for python-urllib3",
				"title": "Title of the patch"
			},
			{
				"category": "description",
				"text": "This update for python-urllib3 fixes the following issues:\n\n- CVE-2023-45803: Fix a request body leak that could occur when\n  receiving a 303 HTTP response (bsc#1216377).\n- CVE-2018-25091: Fixed a potential leak of the Authorization header\n  during a cross-origin redirect (bsc#1216275).\n- CVE-2023-43804: Fixed a potential cookie leak via HTTP redirect if\n  the user manually set the corresponding header (bsc#1215968).\n",
				"title": "Description of the patch"
			},
			{
				"category": "details",
				"text": "SUSE-2023-4352,SUSE-OpenStack-Cloud-9-2023-4352,SUSE-OpenStack-Cloud-Crowbar-9-2023-4352",
				"title": "Patchnames"
			},
			{
				"category": "legal_disclaimer",
				"text": "CSAF 2.0 data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
				"title": "Terms of use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://www.suse.com/support/security/contact/",
			"name": "SUSE Product Security Team",
			"namespace": "https://www.suse.com/"
		},
		"references": [
			{
				"category": "external",
				"summary": "SUSE ratings",
				"url": "https://www.suse.com/support/security/rating/"
			},
			{
				"category": "self",
				"summary": "URL of this CSAF notice",
				"url": "https://ftp.suse.com/pub/projects/security/csaf/suse-su-2023_4352-1.json"
			},
			{
				"category": "self",
				"summary": "URL for SUSE-SU-2023:4352-1",
				"url": "https://www.suse.com/support/update/announcement/2023/suse-su-20234352-1/"
			},
			{
				"category": "self",
				"summary": "E-Mail link for SUSE-SU-2023:4352-1",
				"url": "https://lists.suse.com/pipermail/sle-updates/2023-November/032586.html"
			},
			{
				"category": "self",
				"summary": "SUSE Bug 1215968",
				"url": "https://bugzilla.suse.com/1215968"
			},
			{
				"category": "self",
				"summary": "SUSE Bug 1216275",
				"url": "https://bugzilla.suse.com/1216275"
			},
			{
				"category": "self",
				"summary": "SUSE Bug 1216377",
				"url": "https://bugzilla.suse.com/1216377"
			},
			{
				"category": "self",
				"summary": "SUSE CVE CVE-2018-25091 page",
				"url": "https://www.suse.com/security/cve/CVE-2018-25091/"
			},
			{
				"category": "self",
				"summary": "SUSE CVE CVE-2023-43804 page",
				"url": "https://www.suse.com/security/cve/CVE-2023-43804/"
			},
			{
				"category": "self",
				"summary": "SUSE CVE CVE-2023-45803 page",
				"url": "https://www.suse.com/security/cve/CVE-2023-45803/"
			}
		],
		"title": "Security update for python-urllib3",
		"tracking": {
			"current_release_date": "2023-11-02T16:53:22Z",
			"generator": {
				"date": "2023-11-02T16:53:22Z",
				"engine": {
					"name": "cve-database.git:bin/generate-csaf.pl",
					"version": "1"
				}
			},
			"id": "SUSE-SU-2023:4352-1",
			"initial_release_date": "2023-11-02T16:53:22Z",
			"revision_history": [
				{
					"date": "2023-11-02T16:53:22Z",
					"number": "1",
					"summary": "Current version"
				}
			],
			"status": "final",
			"version": "1"
		}
	},
	"product_tree": {
		"branches": [
			{
				"branches": [
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python-urllib3-1.23-3.25.1.noarch",
								"product": {
									"name": "python-urllib3-1.23-3.25.1.noarch",
									"product_id": "python-urllib3-1.23-3.25.1.noarch"
								}
							},
							{
								"category": "product_version",
								"name": "python3-urllib3-1.23-3.25.1.noarch",
								"product": {
									"name": "python3-urllib3-1.23-3.25.1.noarch",
									"product_id": "python3-urllib3-1.23-3.25.1.noarch"
								}
							}
						],
						"category": "architecture",
						"name": "noarch"
					},
					{
						"branches": [
							{
								"category": "product_name",
								"name": "SUSE OpenStack Cloud 9",
								"product": {
									"name": "SUSE OpenStack Cloud 9",
									"product_id": "SUSE OpenStack Cloud 9",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:suse-openstack-cloud:9"
									}
								}
							},
							{
								"category": "product_name",
								"name": "SUSE OpenStack Cloud Crowbar 9",
								"product": {
									"name": "SUSE OpenStack Cloud Crowbar 9",
									"product_id": "SUSE OpenStack Cloud Crowbar 9",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:suse-openstack-cloud-crowbar:9"
									}
								}
							}
						],
						"category": "product_family",
						"name": "SUSE Linux Enterprise"
					}
				],
				"category": "vendor",
				"name": "SUSE"
			}
		],
		"relationships": [
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-urllib3-1.23-3.25.1.noarch as component of SUSE OpenStack Cloud 9",
					"product_id": "SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch"
				},
				"product_reference": "python-urllib3-1.23-3.25.1.noarch",
				"relates_to_product_reference": "SUSE OpenStack Cloud 9"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-urllib3-1.23-3.25.1.noarch as component of SUSE OpenStack Cloud Crowbar 9",
					"product_id": "SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
				},
				"product_reference": "python-urllib3-1.23-3.25.1.noarch",
				"relates_to_product_reference": "SUSE OpenStack Cloud Crowbar 9"
			}
		]
	},
	"vulnerabilities": [
		{
			"cve": "CVE-2018-25091",
			"ids": [
				{
					"system_name": "SUSE CVE Page",
					"text": "https://www.suse.com/security/cve/CVE-2018-25091"
				}
			],
			"notes": [
				{
					"category": "general",
					"text": "urllib3 before 1.24.2 does not remove the authorization HTTP header when following a cross-origin redirect (i.e., a redirect that differs in host, port, or scheme). This can allow for credentials in the authorization header to be exposed to unintended hosts or transmitted in cleartext. NOTE: this issue exists because of an incomplete fix for CVE-2018-20060 (which was case-sensitive).",
					"title": "CVE description"
				}
			],
			"product_status": {
				"recommended": [
					"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
					"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
				]
			},
			"references": [
				{
					"category": "external",
					"summary": "CVE-2018-25091",
					"url": "https://www.suse.com/security/cve/CVE-2018-25091"
				},
				{
					"category": "external",
					"summary": "SUSE Bug 1216275 for CVE-2018-25091",
					"url": "https://bugzilla.suse.com/1216275"
				}
			],
			"remediations": [
				{
					"category": "vendor_fix",
					"details": "To install this SUSE Security Update use the SUSE recommended installation methods like YaST online_update or \"zypper patch\".\n",
					"product_ids": [
						"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
						"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
					]
				}
			],
			"scores": [
				{
					"cvss_v3": {
						"baseScore": 6.5,
						"baseSeverity": "MEDIUM",
						"vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:N/A:N",
						"version": "3.1"
					},
					"products": [
						"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
						"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"date": "2023-11-02T16:53:22Z",
					"details": "moderate"
				}
			],
			"title": "CVE-2018-25091"
		},
		{
			"cve": "CVE-2023-43804",
			"ids": [
				{
					"system_name": "SUSE CVE Page",
					"text": "https://www.suse.com/security/cve/CVE-2023-43804"
				}
			],
			"notes": [
				{
					"category": "general",
					"text": "urllib3 is a user-friendly HTTP client library for Python. urllib3 doesn't treat the `Cookie` HTTP header special or provide any helpers for managing cookies over HTTP, that is the responsibility of the user. However, it is possible for a user to specify a `Cookie` header and unknowingly leak information via HTTP redirects to a different origin if that user doesn't disable redirects explicitly. This issue has been patched in urllib3 version 1.26.17 or 2.0.5.",
					"title": "CVE description"
				}
			],
			"product_status": {
				"recommended": [
					"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
					"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
				]
			},
			"references": [
				{
					"category": "external",
					"summary": "CVE-2023-43804",
					"url": "https://www.suse.com/security/cve/CVE-2023-43804"
				},
				{
					"category": "external",
					"summary": "SUSE Bug 1215968 for CVE-2023-43804",
					"url": "https://bugzilla.suse.com/1215968"
				}
			],
			"remediations": [
				{
					"category": "vendor_fix",
					"details": "To install this SUSE Security Update use the SUSE recommended installation methods like YaST online_update or \"zypper patch\".\n",
					"product_ids": [
						"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
						"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
					]
				}
			],
			"scores": [
				{
					"cvss_v3": {
						"baseScore": 5.9,
						"baseSeverity": "MEDIUM",
						"vectorString": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N",
						"version": "3.1"
					},
					"products": [
						"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
						"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"date": "2023-11-02T16:53:22Z",
					"details": "moderate"
				}
			],
			"title": "CVE-2023-43804"
		},
		{
			"cve": "CVE-2023-45803",
			"ids": [
				{
					"system_name": "SUSE CVE Page",
					"text": "https://www.suse.com/security/cve/CVE-2023-45803"
				}
			],
			"notes": [
				{
					"category": "general",
					"text": "urllib3 is a user-friendly HTTP client library for Python. urllib3 previously wouldn't remove the HTTP request body when an HTTP redirect response using status 301, 302, or 303 after the request had its method changed from one that could accept a request body (like `POST`) to `GET` as is required by HTTP RFCs. Although this behavior is not specified in the section for redirects, it can be inferred by piecing together information from different sections and we have observed the behavior in other major HTTP client implementations like curl and web browsers. Because the vulnerability requires a previously trusted service to become compromised in order to have an impact on confidentiality we believe the exploitability of this vulnerability is low. Additionally, many users aren't putting sensitive data in HTTP request bodies, if this is the case then this vulnerability isn't exploitable. Both of the following conditions must be true to be affected by this vulnerability: 1. Using urllib3 and submitting sensitive information in the HTTP request body (such as form data or JSON) and 2. The origin service is compromised and starts redirecting using 301, 302, or 303 to a malicious peer or the redirected-to service becomes compromised. This issue has been addressed in versions 1.26.18 and 2.0.7 and users are advised to update to resolve this issue. Users unable to update should disable redirects for services that aren't expecting to respond with redirects with `redirects=False` and disable automatic redirects with `redirects=False` and handle 301, 302, and 303 redirects manually by stripping the HTTP request body.\n",
					"title": "CVE description"
				}
			],
			"product_status": {
				"recommended": [
					"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
					"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
				]
			},
			"references": [
				{
					"category": "external",
					"summary": "CVE-2023-45803",
					"url": "https://www.suse.com/security/cve/CVE-2023-45803"
				},
				{
					"category": "external",
					"summary": "SUSE Bug 1216377 for CVE-2023-45803",
					"url": "https://bugzilla.suse.com/1216377"
				}
			],
			"remediations": [
				{
					"category": "vendor_fix",
					"details": "To install this SUSE Security Update use the SUSE recommended installation methods like YaST online_update or \"zypper patch\".\n",
					"product_ids": [
						"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
						"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
					]
				}
			],
			"scores": [
				{
					"cvss_v3": {
						"baseScore": 4.2,
						"baseSeverity": "MEDIUM",
						"vectorString": "CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
						"version": "3.1"
					},
					"products": [
						"SUSE OpenStack Cloud 9:python-urllib3-1.23-3.25.1.noarch",
						"SUSE OpenStack Cloud Crowbar 9:python-urllib3-1.23-3.25.1.noarch"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"date": "2023-11-02T16:53:22Z",
					"details": "moderate"
				}
			],
			"title": "CVE-2023-45803"
		}
	]
}
